mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,383 @@
+
+
+@io.coerce_to_string_io
+def load_key(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+
+    Returns:
+        KeyData: loaded key data
+
+    """
+    keys = load_key_base(fhandle, resolution)
+    start_times, end_times, key_names = [0], [], [str(keys[0]["key"]).replace("-", "b")]
+    for ii, k in enumerate(keys):
+        if str(k["key"]).replace("-", "b") != key_names[-1]:
+            end_times.append(keys[ii]["time"] - 1)
+            start_times.append(keys[ii]["time"])
+            key_names.append(str(keys[ii]["key"]).replace("-", "b"))
+    end_times.append(keys[-1]["time"])
+    return KeyData(np.array([start_times, end_times]).astype(float).T, key_names)
+
+
+@io.coerce_to_string_io
+def load_key_music21(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+
+    Returns:
+        List[dict]: musical key data and relative time (offset (Music21Object.offset) * resolution).
+
+    """
+    return load_key_base(fhandle, resolution)
+
+
+@io.coerce_to_string_io
+def load_midi_path(fhandle: TextIO):
+    """Load path to midi file of haydn op20 musical piece
+
+    Args:
+        fhandle (str or file-like): path to score file
+
+    Returns:
+        str: midi file path
+
+    """
+    midi_path = os.path.splitext(fhandle.name)[0] + ".midi"
+    if not os.path.exists(midi_path):
+        score, _ = split_score_annotations(fhandle)
+        score.write("midi", fp=midi_path)
+    return midi_path
+
+
+@io.coerce_to_string_io
+def load_roman_numerals(fhandle: TextIO, resolution=28):
+    """Load haydn op20 roman numerals data from a file
+
+    Args:
+        fhandle (str or file-like): path to roman numeral annotations
+
+    Returns:
+        List[dict]: musical roman numerals data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        figure = rn.figure
+        annotations.append({"time": time, "roman_numeral": figure})
+    return annotations
+
+
+def load_chords_base(fhandle: TextIO, resolution: int = 28):
+    """Load haydn op20 chords data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to chord annotations
+
+    Returns:
+        List[dict]: musical chords data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        chord = rn.pitchedCommonName
+        annotations.append({"time": time, "chord": chord})
+    return annotations
+
+
+@io.coerce_to_string_io
+def load_chords(fhandle: TextIO, resolution: int = 28):
```

Comment:
what does resolution control? What are its units? Can you describe it i the Args of the docstring?

Reply:
>> The resolution is the number of pulses, or ticks, per quarter note (PPQ) and determines the precision with which you will be able to view and edit the MIDI data. The higher the resolution, the higher the precision. The resolution should be chosen depending on the application or sequencer with which the MIDI file will be used though, because certain applications and sequencers may not be able to handle certain resolutions.

Reply:
The the argument is going to be
`resolution (int): the number of pulses, or ticks, per quarter note (PPQ)`

Reply:
done